snowfakery/data_generator_runtime.py

```diff
@@ -93,18 +90,17 @@ def generate_id(self, table_name: str) -> int:
     def __getitem__(self, table_name: str) -> int:
         return self.last_used_ids[table_name]
 
+    def __getstate__(self):
+        return {"last_used_ids": dict(self.last_used_ids)}
+
     def __setstate__(self, state):
         self.last_used_ids = defaultdict(lambda: 0, state["last_used_ids"])
 
 
-yaml.SafeDumper.add_representer(defaultdict, yaml.SafeDumper.represent_dict)
+SnowfakeryDumper.add_representer(defaultdict, SnowfakeryDumper.represent_dict)
```

Comment:
Would it make sense to centralize these where SnowfakeryDumper is defined instead of scattered throughout the code?

Reply:
I thought about it, but felt like it would end up being a sort of hub-and-spoke which might cause circular dependencies and otherwise-unrelated imports centralized in one file. defaultdict is a bit of a special case because its a standard function but for example PluginResult is a "concept" that makes the most sense in the context of the plugins module. There are only 4 of them though, so I don't feel strongly one way or the other.